pkg/storage/snapshot/restore.go

```diff
@@ -49,6 +49,8 @@ import (
 const (
 	RestoreNameAnnotation = "restore.kubevirt.io/name"
 
+	vmRestoreFinalizer = "snapshot.kubevirt.io/vmrestore-protection"
```

Comment:
Just a note about finalizers, we used to have a bug where we forget to remove one, so the resource could be left dangling. I don't see how it's possible with this PR but just for awareness https://github.com/kubevirt/kubevirt/pull/11241/commits/6ad91329f2e72aaa6b6c961fdf38496052602621

Reply:
Should make sure finalizers are removed here as well, for example: https://github.com/kubevirt/kubevirt/blob/main/hack/cluster-clean.sh#L49-L54